### PR TITLE
Update cmake windows docs

### PR DESCRIPTION
**Pull request by dave3d**

### CODE REVIEW

File: Examples/CppCMake/Documentation.rst [hunk 1]
```diff
@@ -17,86 +19,105 @@ Here is our SimpleITK C++ example file **sitk_example.cxx**.
...
+:doc:`building`, then these required directories can be found in the SuperBuild
+location.  The SimpleITK_DIR variable should refer to the directory
+SimpleITK-build within the SuperBuild location, and the ITK_DIR variable should
+refer to the directory ITK-build in the SuperBuild location.
 
 Here are what the settings in a CMakeCache.txt file should look like::
 
     //The directory containing a CMake configuration file for ITK.
-    ITK_DIR:PATH=/home/dchen/SuperBuild/ITK-build
+    ITK_DIR:PATH=/home/dchen/SimpleITK-build/ITK-build
```
Comment (zivy) [inline Examples/CppCMake/Documentation.rst:50]: Possibly replace "/home/dchen/" with the more generic "path_to_simpleitk_build_directory/"?

**Comment by blowekamp** [comment]
Do you think the new Windows build page should be a top-level to somehow under the existing "build" page?

File: Examples/CppCMake/Documentation.rst [hunk 2]
```diff
@@ -17,86 +19,105 @@ Here is our SimpleITK C++ example file **sitk_example.cxx**.
...
 
 
-Using CMake on Windows
-######################
+Using CMake-gui on Windows
+##########################
 
-The Windows version of CMake includes a graphical-user-interface (GUI) version of the program,
-in addition to the standard command-line CMake.  Here is what the CMake GUI looks like:
+The Windows version of CMake includes a graphical-user-interface (GUI) version
```
Comment (zivy) [inline Examples/CppCMake/Documentation.rst:74]: The GUI is not specific to windows (I use it on OSX). Possibly say "In addition to the standard command-line CMake, a graphical-user-interface (GUI) version of CMake is also available.

File: Examples/CppCMake/Documentation.rst [hunk 3]
```diff
@@ -17,86 +19,105 @@ Here is our SimpleITK C++ example file **sitk_example.cxx**.
...
-the proper platform.
-
-.. image:: ../images/cmake-win-5.png
-  :width: 500
-  :alt: CMake Build Version
+  :alt: Setting ITK_DIR
+
+After clicking **Configure**, CMake will attempt to find the SimpleITK library.
+That can been set in the :code:`SimpleITK_DIR` variable, as seen in the above
+left image.  CMake is looking for the directory containing the
```
Comment (zivy) [inline Examples/CppCMake/Documentation.rst:102]: been -> be

File: Examples/CppCMake/Documentation.rst [hunk 4]
```diff
@@ -17,86 +19,105 @@ Here is our SimpleITK C++ example file **sitk_example.cxx**.
...
+
+.. image:: ../images/windows-build4.png
+  :width: 90%
+  :alt: Visual Studio
+
+After opening :code:`ALL_BUILD.vcvproj` in Visual Studio, the user should
+select a build type in Visual Studio.  We recommend choosing **Release** for
+performance reasons.  Then the user should select the **x64** build platform,
+not the unsupported default, x86.  Many man-hours and cpu cycles have been
+wasted building SimpleITK without setting the proper platform.
```
Comment (zivy) [inline Examples/CppCMake/Documentation.rst:123]: So sad and so true.

File: docs/source/buildingWindows.rst [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+of the program, in addition to the standard command-line CMake.  Here is what
+the CMake GUI looks like:
+
+.. image:: ../images/cmake-win-1.png
+  :width: 400
+  :alt: Windows CMake GUI
+
+In the above example, the source directory is 'C:/SimpleITK', and the
+build directory is 'C:/SimpleITK-build'.  It is best to place the source
+and build directories at the root of a volume to minimize path name lengths.
```
Comment (zivy) [inline docs/source/buildingWindows.rst:19]: Possibly explicitly say that windows has a default max path length of 260 characters and ITK has long paths which can cause build issues? https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd

File: docs/source/buildingWindows.rst [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+The next steps are the configuration and generation processes.  By clicking
+the **Configure** button, cmake reads the CMakeLists.txt file and sets up all
+the variables in the CMake cache (CMakeCache.txt).  Then when the **Generate**
+button is clicked, CMake creates the Visual Studio build system.
+
+The left image of the above figure is the result of the configure process, and
+on the right is the result of the generation process.
+
+Note for the **CMAKE_CXX_FLAGS** and **CMAKE_C_FLAGS** variables, we have
```
Comment (zivy) [inline docs/source/buildingWindows.rst:39]: Any way to highlight this? It isn't a "nice to have" it's a must have. Without the /MP flag the developer will die of old age before the SimpleITK build finishes.

**Review by zivy — APPROVED** [review]
Some easily addressed comments.

File: docs/source/buildingWindows.rst (hunk 6 above)
Comment (dave3d) [reply docs/source/buildingWindows.rst:39]: shouldn't we be setting that automatically?

**Comment by dave3d** [comment]
> Do you think the new Windows build page should be a top-level to somehow under the existing "build" page?

You mean a tab under the Building SimpleITK page?  I could do that.

File: docs/source/buildingWindows.rst (hunk 6 above)
Comment (blowekamp) [reply docs/source/buildingWindows.rst:39]: In general /MP does not behave well with CMake super build projects as there is not global process manager, each sub-project runs /MP with out regards to the parent. Too many builds can run on CI services and memory issues can occur.

Comment (zivy) [reply docs/source/buildingWindows.rst:39]: Got it. So we leave the instructions as is, and it is up to the person compiling it on windows to do the right thing.

File: Examples/CppCMake/Documentation.rst (hunk 2 above)
Comment (dave3d) [reply Examples/CppCMake/Documentation.rst]: done

**Pull request merged**